src/components/widgets-common/data-not-found/data-not-found.directive.coffee

```diff
@@ -5,26 +5,38 @@ module.directive('commonDataNotFound', ($templateCache, $log, $http, ImpacAssets
     restrict: 'A',
     scope: {
       widgetEngine: '='
+      widgetWidth: '='
       onDisplayAlerts: '&'
     },
     link: (scope, element) ->
       scope.content = ImpacTheming.get().dataNotFoundConfig
-      baseDir = ImpacAssets.get('dataNotFound')
+      baseDir       = ImpacAssets.get('dataNotFound')
+      usingDefaults = false
+      image         = _.find(element.children().first().children(), (elem) -> elem.id == 'not-found-bg')
+
+      imagePath     = ''
+
+      # When providing custom images
       if scope.widgetEngine and baseDir.length > 0
         # checks for trailing slash and corrects.
         dir = baseDir.split('')
         dir = if dir[dir.length - 1] != '/'
         then dir.concat('/').join('')
         else dir.join('')
 
-        image = _.find element.children().first().children(), (elem) ->
-          elem.id == 'not-found-bg'
+        imagePath = dir + scope.widgetEngine + '.png'
+
+      # When providing no custom images, uses library defaults.
+      else
+        usingDefaults = true
+        imagePath = 'dist/images/widget-bg-width-' + scope.widgetWidth + '.png'
 
-        image.onerror = ->
-          $log.warn("Missing data-not-found image for #{scope.widgetEngine}")
-          image.remove() if image?
+      image.onerror = ->
+        missingImageLocationMsg = if usingDefaults then 'library defaults' else scope.widgetEngine
+        $log.warn("Missing data-not-found image for #{missingImageLocationMsg}")
```

Comment:
ok for the case 2.,
but the case 1. should fallback to the default image. Otherwise, when you start uploading custom screenshots for a few widgets, you're forced to add one for each of the widgets.